prusti-specs/src/lib.rs

```diff
@@ -445,12 +446,56 @@ pub fn refine_trait_spec(_attr: TokenStream, tokens: TokenStream) -> TokenStream
     }
 }
 
+pub fn export_spec(attr: TokenStream, tokens: TokenStream) -> TokenStream {
+    let item: syn::Item = handle_result!(syn::parse2(tokens));
+    let item_span = item.span();
+    let path: syn::Path = handle_result!(export_spec_rewriter::parse_valid_export(attr));
+    let macros = handle_result!(export_spec_rewriter::rewrite_export_spec(&path, &item));
+    parse_quote_spanned!(item_span => 
+        #item
+        #macros
+    )
+}
+
+pub fn prusti_use(tokens: TokenStream) -> TokenStream {
+    let path: syn::Path = handle_result!(syn::parse2(tokens.into()));
+    let macro_path = handle_result!(prusti_use_macro(path));
+    parse_quote_spanned!(macro_path.span()=> #macro_path!();) 
+}
+
+fn prusti_use_macro(path: syn::Path) -> syn::Result<syn::Path> {
+    let mut macro_path = syn::Path {
+        leading_colon: None,
+        segments: syn::punctuated::Punctuated::new(),
+    };
+    let mut mod_path_for_hashing = syn::Path {
+        leading_colon: None,
+        segments: syn::punctuated::Punctuated::new(),
+    };
+
+    let mut seg_iter = path.segments.clone().into_iter();
+
+    if let Some(crate_seg) = seg_iter.next() {
```

Comment:
Why not `first` instead of cloning and creating an iterator?

Reply:
Oh because at here if our path do have the first and the last segment then I still need to get the segments after the first one, because first one is the crate: `crate_name::mod::fn`.
The path to generate hash should be `mod::fn`, therefore I will need to have the iterator step through the first one anyway